dquartic/dia-nn to massdash feature extraction/Feature_extraction.py

```diff
@@ -0,0 +1,397 @@
+        
+        plt.figure(figsize=(10, 5))
+        plt.bar(id_counts["Run"], id_counts["Identifications"], color="skyblue")
+        plt.xticks(rotation=45, ha="right")
+        plt.xlabel("File")
+        plt.ylabel("Number of Identifications")
+        plt.title("Identifications per File (1% Precursor FDR)")
+        plt.grid(axis="y", linestyle="--", alpha=0.7)
+        plt.tight_layout()
+        
+        plot_path = os.path.join(self.output_dir, "identifications_per_file.png")
+        plt.savefig(plot_path)
+        logging.info(f"Plot saved to {plot_path}")
+        return plot_path
+
+    def initialize_loader(self):
+        """
+        Initialize the MassDASH data loader to access mzML files.
+
+        The loader requires the TSV files produced by convert_parquet_to_tsv.
+        """
+        logging.info("Initializing MassDASH loader...")
+        report_tsv_path = os.path.join(os.path.dirname(self.output_dir), 'report.tsv')
+        gen_lib_modified_tsv_path = os.path.join(os.path.dirname(self.output_dir), 'gen_lib_modified.tsv')
+        
+        self.loader = MzMLDataLoader(
+            dataFiles=self.mzml_path,
+            rsltsFile=report_tsv_path,
+            libraryFile=gen_lib_modified_tsv_path
+        )
+        logging.info(f"MassDASH loader initialized with mzML path: {self.mzml_path}")
+
+    def configure_extraction(self):
+        """
+        Configure the extraction parameters for feature extraction.
+
+        Sets the retention time window and m/z tolerance.
+        """
+        logging.info("Configuring extraction parameters...")
+        self.extraction_config = TargetedDIAConfig()
+        self.extraction_config.rt_window = self.rt_window
+        self.extraction_config.mz_tol = self.mz_tol
+        logging.info(f"Extraction configured with RT window: {self.rt_window}s, m/z tolerance: {self.mz_tol} ppm")
+
+    def run_batch_processing(self):
+        """
+        Process peptides in batches to extract MS1 and MS2 features using GPU acceleration.
+
+        Steps:
+            1. Load unique peptide-charge pairs from the report.
+            2. Split the peptides into batches based on the configured batch size.
+            3. For each peptide, extract features via the MassDASH loader.
+            4. Convert the feature data to a GPU DataFrame (cuDF) for accelerated filtering.
+            5. Extract and rename MS1 (precursor) and MS2 (fragment) features.
+            6. Save each batch as CSV files and combine all batches into final output files.
+
+        Returns:
+            dict: Summary of processing including total peptides and output file paths.
+        """
+        logging.info(f"Loading peptides from {self.report_path}")
+        report_df = pd.read_parquet(self.report_path)
+        unique_peptides = report_df[['Modified.Sequence', 'Precursor.Charge']].drop_duplicates()
+        logging.info(f"Total unique peptides: {len(unique_peptides)}")
```

Comment:
```suggestion
        logging.info(f"Total unique precursors: {len(unique_peptides)}")
```

Peptide is just the sequence i.e. `PEPTIDK`. Precursor is the ionized peptide sequence (i.e. has a charge state, `PEPTIDK_2`, `PEPTIDK_3`, `ASTYK_2` etc..)

Reply:
Thanks for improving my understanding. I've updated the logging statement as suggested.